### PR TITLE
Feat: Add seeding to db

### LINKED ISSUE #53: Feature: Add a seeding to db
**Is your feature request related to a problem? Please describe.**
whenever new developer set the repo they have to create their own data to test for any new feature

**Describe the solution you'd like**
to tackle this what usually does is a seed file is created as its name suggests on the initial run of the server we will seed some demo data into our database so that developers don't have to create test data.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by pantharshit007**
## Description

**Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.**

Closes #53 

### Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please specify):

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [] New and existing unit tests pass locally with my changes

### Screenshots (if applicable)

**Please include any relevant screenshots or images that help explain the changes made.**

added seed function with little bit of cleanup of Mongo connection string.

image

_DB: Mongo_
image

image

*Do add the hacktober tag*

### CODE REVIEW

File: src/db/dbConnection.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    } catch (err) {
+        console.log("> Error connecting to DB:");
+        console.error(err);
+        process.exit(1);
+    }
+
+}
+
+async function seedDB() {
```
Comment (dead8309) [inline src/db/dbConnection.js:20]: Seeding script should be in a standalone file

Comment (pantharshit007) [reply src/db/dbConnection.js]: done you can check!

File: src/db/dbConnection.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import mongoose from "mongoose";
+import "dotenv/config";
+import bcrypt from "bcryptjs";
+import User from "../models/User.js";
+import Blogs from "../models/Blogs.js";
+
+async function connectDatabase() {
+    try {
+        await mongoose.connect(process.env.MONGODB_URL);
```
Comment (dead8309) [inline src/db/dbConnection.js:9]: Add the connection properties which were used before 
```
{
    useNewUrlParser: true,
    useUnifiedTopology: true,
 }
```

File: src/index.js [hunk 3]
```diff
@@ -54,30 +56,22 @@ app.get("/health", (_, res) => {
...
   console.log(
-    `Swagger API documentation available at http://localhost:${PORT}/api-docs`
+    `> Swagger API documentation available at http://localhost:${PORT}/api-docs`
   );
 });
+
+// Connect database to app
+connectDatabase();
+// seed the database
+seedDB();
```
Comment (dead8309) [inline src/index.js:77]: Remove the `seedDB()` call from here and instead add a script in `package.json` 
```json
"db:seed": "node src/db/seed.js"
```

File: src/controllers/blog.js
Comment (dead8309) [inline src/controllers/blog.js:1]: This file is not related to the scope of this pr, can you split this change into a seperate pr ?

File: src/index.js [hunk 4]
```diff
@@ -54,30 +56,22 @@ app.get("/health", (_, res) => {
 });
 
 // Routes
-app.use("/auth", authRoute);
-app.use("/", blogRoute);
+app.use("/api/auth", authRoute);
+app.use("/api", blogRoute);
```
Comment (dead8309) [inline src/index.js:60]: The change above is not related to this pr, maybe split it into a new one

File: src/routes/auth.js [hunk 5]
```diff
@@ -62,13 +62,13 @@ router.post("/login", validateRequest(loginSchema), (req, res) => {
...
  *               - password
- *               - name
+ *               - username
  *             properties:
  *               email:
  *                 type: string
  *               password:
  *                 type: string
- *               name:
+ *               username:
```
Comment (dead8309) [inline src/routes/auth.js:71]: Open a new issue and pr for this aswell

File: src/db/dbConnection.js (hunk 2 above)
Comment (pantharshit007) [reply src/db/dbConnection.js:9]: they are soon to be deprecated npm gave me the warning and its running fine without it

File: src/index.js (hunk 3 above)
Comment (pantharshit007) [reply src/index.js:77]: ok

File: src/db/dbConnection.js (hunk 2 above)
Comment (dead8309) [reply src/db/dbConnection.js:9]: Okay then

File: src/controllers/blog.js
Comment (pantharshit007) [reply src/controllers/blog.js:1]: these are small changes which are required since we are creating a new directory and files
and the other change was a simple type in mongo URL

It's fine but the last call is yours, tell me if I still should neglect all the changes in this file.

Comment (dead8309) [reply src/controllers/blog.js:1]: Yup reset these changes and create a separate pr for that

File: src/models/Blogs.js [hunk 6]
```diff
@@ -8,4 +8,7 @@ const blogsSchema = new mongoose.Schema({
   timestamps: true
 });
 
+// compound index to ensure that a user can't have multiple blogs with the same title
+blogsSchema.index({ user: 1, title: 1 }, { unique: true });
```
[the remark this answers is not in the document]
Comment (pantharshit007) [reply src/models/Blogs.js:12]: this is not needed now since we are making seed a stand alone command now.

[the remark this answers is not in the document]
Comment (dead8309) [reply src/models/Blogs.js:12]: Great!

[the remark this answers is not in the document]
Comment (dead8309) [reply src/models/Blogs.js:12]: Remove this change then

File: src/db/seed.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        console.log("> User created:", "Email:", user.email, "Pass:", password);
+        console.log("> Blog created (1):", blog1.upsertedId || blog1.acknowledged);
+        console.log("> Blog created (2):", blog2.upsertedId || blog2.acknowledged);
+    } catch (error) {
+        console.log("> Failed to seed DB:");
+        console.error(error);
+    }
+}
+
+export default seedDB;
```
Comment (dead8309) [inline src/db/seed.js:54]: call seedDB here directly instead of exporting it

Comment (pantharshit007) [reply src/db/seed.js:54]: It's just a statement that I missed to remove I am directly call this using `npm run db:seed`

Comment (dead8309) [reply src/db/seed.js:54]: How will it run without invoking the function ? You *have* to call ```seedDB()``` here

Comment (pantharshit007) [reply src/db/seed.js:54]: oh my bad! I will fix that

**Review by dead8309 — APPROVED** [review]

**Comment by dead8309** [comment]
Thanks for your contribution 🚀
